spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageUtils.java

```diff
@@ -122,8 +122,8 @@ else if (messageAttribute.getValue().getDataType()
 				messageHeaders.put(messageAttribute.getKey(),
 						getNumberValue(messageAttribute.getValue()));
 			}
-			else if (MessageAttributeDataTypes.BINARY
-					.equals(messageAttribute.getValue().getDataType())) {
+			else if (messageAttribute.getValue().getDataType()
+					.startsWith(MessageAttributeDataTypes.BINARY)) {
```

Comment:
let's change it to `MessageAttributeDataTypes. BINARY.startsWith(messageAttribute.getValue().getDataType())`

Reply:
Here is the same as in comment above.